deployment/common/changeset/apply.go

```diff
@@ -0,0 +1,91 @@
+package changeset
+
+import (
+	"context"
+	"fmt"
+
+	mapset "github.com/deckarep/golang-set/v2"
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/gethwrappers"
+	jobv1 "github.com/smartcontractkit/chainlink-protos/job-distributor/v1/job"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+)
+
+type ChangesetApplication struct {
+	Changeset func(e deployment.Environment, config any) (deployment.ChangesetOutput, error)
+	Config    any
+}
+
+func WrapChangeSet[C any](fn deployment.ChangeSet[C]) func(e deployment.Environment, config any) (deployment.ChangesetOutput, error) {
+	return func(e deployment.Environment, config any) (deployment.ChangesetOutput, error) {
+		var zeroC C
+		if config != nil {
+			c, ok := config.(C)
+			if !ok {
+				return deployment.ChangesetOutput{}, fmt.Errorf("invalid config type, expected %T", c)
+			}
+			return fn(e, config.(C))
+		}
+
+		return fn(e, zeroC)
+	}
+}
+
+// ApplyChangesets applies the changeset applications to the environment and returns the updated environment.
+func ApplyChangesets(ctx context.Context, e deployment.Environment, timelocksPerChain map[uint64]*gethwrappers.RBACTimelock, changesetApplications []ChangesetApplication) (deployment.Environment, error) {
```

Comment:
I see, ChangesetApplication here can have a bunch of different config types, so can't be generic